colossalai/utils/activation_checkpoint.py

```diff
@@ -32,7 +34,12 @@ def forward(ctx, run_function, *args):
         tensor_inputs = []
```

Comment:
You missed an import procedure.
The forward can be executed during BWD of the model (grad checkpointing).
args should be guaranteed on cuda device before passing to run_function(*args).
I believe in some cases the args may resident on CPU.
https://github.com/microsoft/DeepSpeed/blob/013df6a9e8aef5d20ae0cf059c5fbd81cf90b694/deepspeed/runtime/activation_checkpointing/checkpointing.py#L569

Reply:
ok, I will try to add the strategy for this case